File: FitnessApp-master/Services/UserService.cs
using fitnessData.Auth;
using fitnessData.Utils;
using fitnessapp.Models;
using fitnessapp.Models.RequestModels;
using fitnessapp.Models.ResponseModels;
using System.Linq;
using System;

namespace fitnessapp.Services
{
    public class UserService
    {
        private AuthDbContext _userContext;

        public UserService(AuthDbContext userContext)
        {
            _userContext = userContext;
        }

        public UserDTO Get(int id)
        {
            var user = _userContext.Users.Find(id);

            return UserDTO.Create(user);
        }

        public ResponseVM CreateNew(RegisterRequest request)
        {

            if (_userContext.Users.Any(u => u.Email == request.Email))
            {
                return new ResponseVM(false);
            }

            var passwordhash = CryptoUtils.GetMD5Hash(request.Password);

            var newUser = new ApplicationUser(request.Email, passwordhash);

            var user = _userContext.Add(newUser).Entity;
            _userContext.SaveChanges();

            user = CreateUserToken(user);

            return new ResponseVM(true, UserDTO.Create(user));
        }

        public ResponseVM Delete(int id)
        {
            if (id == 1)
            {
                return new ResponseVM(false);
            }

            var user = _userContext.Users.Find(id);

            if (user == null)
            {
                return new ResponseVM(false);
            }
            _userContext.Users.Remove(user);
            _userContext.SaveChanges();

            return new ResponseVM(true);
        }

        public ResponseVM Update(UserDTO user)
        {
            if (user.IsTestUser) {
                return new ResponseVM(false, user);
            }
            var updatedUser = _userContext.Users.Find(user.Id);
            updatedUser.Email = user.Email;
            updatedUser.UserName = user.UserName;
            _userContext.Users.Update(updatedUser);
            var res = _userContext.SaveChanges();
            return new ResponseVM(true, UserDTO.Create(updatedUser));
        }

        public ApplicationUser CreateUserToken(ApplicationUser user)
        {
            var newToken = CryptoUtils.CreateToken(user.Id);
            user.Token = newToken;
            _userContext.SaveChanges();
            return user;
        }

    }
}

File: FitnessApp-master/ClientApp/src/Services/UserService.js
import axios from 'axios';
import ResponseVM from '../Models/Messages/ResponseVM';
import User from '../Models/User';
import constants from '../Utils/stringConstants';

export default class UserService {

    static login(email, password, receiveresponse, errorcallback) {

        axios.post('auth/login', {
            "Email": email,
            "Password": password,
            "ReturnUrl": "",
            "Token": ""
        })
            .then(function (res) {
                let user = "";

                if (res.data.requestSuccess) {
                    user = new User(res.data.user.id, res.data.user.userName,
                        res.data.user.email, res.data.user.token);

                    let response = new ResponseVM(
                        res.data.requestSuccess,
                        user
                    );
                    receiveresponse(response);
                }
                else {
                    errorcallback();
                }
            })
            .catch(function (error) {
                console.log("error occured: ", error);
            });
    }

    static register(email, password, receiveresponse, errorcallback) {

        axios.post('auth/register', {
            "Email": email,
            "Password": password,
            "ReturnUrl": "",
            "Token": ""
        })
            .then(function (res) {

                let newuser = "";

                if (res.data.requestSuccess) {
                    newuser = new User( res.data.user.id, res.data.user.userName,
                                        res.data.user.email, res.data.user.token);

                    let response = new ResponseVM(
                        res.data.requestSuccess,
                        newuser
                    );
                    receiveresponse(response);
                }
                else {
                    errorcallback();
                }
            })
            .catch(function (error) {
                console.log("error occured: ", error);
            });
    }

    static logout(user, receiveresponse) {

        axios.post('auth/logout', {
            "User": user,
            "ReturnUrl": "",
            "Token": ""
        })
            .then(function (res) {
                let response = new ResponseVM(res.data.requestSuccess);

                receiveresponse(response);
            })
            .catch(function (error) {
                console.log("error occured: ", error);
            });
    }

    static update(user, receiveresponse, callback) {

        axios.post('account/update', {
            "Id": user.id,
            "UserName": user.username,
            "Email": user.email,
            "Token": user.token
        })
            .then(function (res) {
                let newuser = ""

                if (res.data.requestSuccess) {
                    newuser = new User(res.data.user.id, res.data.user.userName,
                        res.data.user.email, res.data.user.token);


                    var response = new ResponseVM(
                        res.data.requestSuccess,
                        newuser
                    );

                    callback(constants.success);
                    receiveresponse(response);
                } else {
                    callback(constants.failed)
                }
            })
            .catch(function (error) {
                console.log("error occured: ", error);
            });

    }

    static delete(user, receiveresponse) {

        axios.delete('account/delete', { params: { id: user.id } })
            .then(function (res) {


                var response = new ResponseVM(res.data.requestSuccess);
        
                receiveresponse(response);
            })
            .catch(function (error) {
                console.log("error occured: ", error);
            });
    }
}